packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -99,6 +95,41 @@ export class SidebarComponent extends BaseContainer {
 		await this.page.waitForSelector( `${ selectors.layout }.focus-content` );
 	}
 
+	/**
+	 * Waits for the sidebar menu items to finish loading.
+	 *
+	 * The bounding box of the Current Site card (located at top of the sidebar) is compared
+	 * every 250ms to determine whether loading has completed.
+	 *
+	 * On some sites, typically atomic, the sidebar undergoes multiple mutations of its structure:
+	 * 	1. Shell of the sidebar gets created and remains static throughout the entire loading process.
+	 * 	2. First version of the sidebar gets created and has only the original basic menu items.
+	 * 	3. The mostly final version of the sidebar gets created. The parent element looks exactly the same
+	 * 		but it is a new node that is attached in the DOM to replace the old one. This final version has
+	 * 		all the menus from different plugins.
+	 *	4. Offers and notices are loaded in a banner that pop in on the top of the sidebar. This does not
+	 *		detach the parent sidebar, or any of the other elements, from the DOM. However, they do shift
+	 *		down slightly on the page.
+	 *
+	 * The bounding box of the Current Site card is one of the last elements to stabilize in the loading
+	 * process and its stability is a good indicator of the page stability.
```

Comment:
I've borrowed and simplified @dpasque 's findings from [this comment](https://github.com/Automattic/wp-calypso/pull/54987#issuecomment-889230498) for the docstring.